packages/slice-machine/lib/builders/CustomTypeBuilder/SliceZone/index.tsx

```diff
@@ -110,7 +110,7 @@ const SliceZone: React.FC<SliceZoneProps> = ({
         onRemoveSharedSlice(key);
       });
     }
-  }, [notFound]);
+  }, [JSON.stringify(notFound)]);
```

Comment:
You should be able to use `useMemo` line 103 instead of this to prevent a rerender